traffic_ops/traffic_ops_golang/cdni/shared.go

```diff
@@ -518,7 +531,14 @@ func checkBearerToken(bearerToken string, inf *api.APIInfo) (string, error) {
 		return "", errors.New("invalid token - incorrect dcdn")
 	}
 	if ucdn == "" {
-		return "", errors.New("invalid token - empty ucdn field")
+		if inf.User.RoleName == tc.AdminRoleName {
+			ucdn = inf.Params["ucdn"]
+			if ucdn == "" {
+				return "", errors.New("admin level ucdn requests require a ucdn query parameter")
+			}
+		} else {
+			return "", errors.New("invalid token - empty ucdn field")
+		}
```

Comment:
Do we need to check if this ucdn claim from the JWT cookie matches the user's stored `ucdn` value?

Reply:
i dont think we need to since we control all sides of the equation but i can add it in to be extra safe. people shouldnt be changing ucdns really and if they do then they would still see their old ucdn until they log out and log back in or the token expires in 6 hours. i was trying to avoid an extra DB call but might as well be extra safe